Doppler.BigQueryMicroservice/Controllers/BigQueryController.cs

```diff
@@ -41,14 +58,38 @@ public async Task<ActionResult> GetAllowedEmails(string accountName)
         [HttpPut("/big-query/{accountName}/allowed-emails")]
         public async Task<ActionResult> SaveAllowedEmails(string accountName, AllowedEmails model)
         {
-            var user = await _userRepository.GetUserByEmail(accountName);
-            if (user == null)
+            try
             {
-                return NotFound();
+                bool notified = false;
+                var user = await _userRepository.GetUserByEmail(accountName);
+                if (user == null)
+                {
+                    return NotFound();
+                }
+
+                var results = await _userAccessByUserRepository.MergeEmailsAsync(user.IdUser, model.Emails);
+                var newEmails = results.Where(a => a.Action == "INSERT").Select(e => e.Email).ToList();
+                var template = _emailSettings.Value.BigQueryInvitationTemplateId[model.Languaje.ToLower()];
+                if (newEmails.Count > 0)
+                {
+                    try
+                    {
+                        notified = await _emailSender.SendWithTemplateAsync(template, newEmails, model.Emails);
+                    }
+                    catch (System.Exception ex)
+                    {
+                        _logger.LogError(ex, "Error in method SendWithTemplateAsync, service {service}", nameof(EmailSender));
+                        notified = false;
+                    }
+                }
+
+                return Ok(new { saved = true, notified = notified });
+            }
+            catch (System.Exception ex)
+            {
+                _logger.LogError(ex, "Error in method SaveAllowedEmails, controller {controller}", nameof(BigQueryController));
```

Comment:
Very, very, very minor and arguable.

But in my opinion, these log errors should be more human-friendly. Also, the name of the class is not required because we already have that metadata available.

For example:

```suggestion
                _logger.LogError(ex, "Error saving allowed emails");
```